### PR TITLE
fix(backend): enforces lower casing of emails

### PR DESCRIPTION
**Pull request by YazeedLoonat**
## Issue
https://github.com/bloom-housing/bloom/issues/1912

Addresses #issue
- [x] This change addresses the issue in full

## Description
Addresses Issue #1912  enforcing lower casing of incoming emails when:
new users are created, a look up of a user by email address happens, during user confirmation, and user log in

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] This change requires a SQL Script

## How Can This Be Tested/Reviewed?
Create a new user with some combination of upper and lower case letters in the email address
attempt to log in as that user but only input the email address in lower case (no upper case characters) you should be able to log in as if the cases matched

if you have DB access, peep the new record and verify the email stored in the db is lower cased

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [x] I have updated the changelog to include a description of my changes
- [x] I have run `yarn generate:client` if I made backend changes
- [ ] I have exported any new pieces in ui-components

### CODE REVIEW

File: backend/core/test/user/user.e2e-spec.ts [hunk 1]
```diff
@@ -570,4 +570,27 @@ describe("Applications", () => {
...
+      middleName: "Mid",
+      lastName: "Last",
+      dob: new Date(),
+    }
+    const mockWelcome = jest.spyOn(testEmailService, "welcome")
+    const res = await supertest(app.getHttpServer())
+      .post(`/user`)
+      .set("jurisdictionName", "Alameda")
+      .send(userCreateDto)
+    expect(mockWelcome.mock.calls.length).toBe(1)
```
Comment (pbn4) [inline backend/core/test/user/user.e2e-spec.ts:590]: Checking whether mockWelcome has been called is not needed here.

File: backend/core/src/migration/1633557587028-user-email-lower-case.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { MigrationInterface, QueryRunner } from "typeorm"
+
+export class userEmailLowerCase1633557587028 implements MigrationInterface {
+  name = "userEmailLowerCase1633557587028"
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
```
Comment (pbn4) [inline backend/core/src/migration/1633557587028-user-email-lower-case.ts:6]: @seanmalbert @YazeedLoonat  Do we only want to change login related email handling here?

We have more columns like %email% in our DB schema:
```
bloom=# SELECT t.table_schema, t.table_name, c.column_name fROM information_schema.tables t INNER JOIN information_schema.columns c on c.table_name = t.table_name and c.table_schema = t.table_schema WHERE c.column_name LIKE '%email%' and t.table_schema not in ('information_schema', 'pg_catalog') and t.table_type = 'BASE TABLE' order by t.table_schema;
 table_schema |    table_name     |     column_name
--------------+-------------------+---------------------
 public       | user_accounts     | email
 public       | listings          | leasing_agent_email
 public       | household_member  | no_email
 public       | applicant         | email_address
 public       | applicant         | no_email
 public       | alternate_contact | email_address
 public       | household_member  | email_address
```

Shouldn't we update those too? E.g. when detecting duplicates in applications (email rule set) by lower casing them we also prevent some problems with missing duplicates.

File: backend/core/test/user/user.e2e-spec.ts [hunk 3]
```diff
@@ -570,4 +570,27 @@ describe("Applications", () => {
...
+    expect(res.body).toHaveProperty("id")
+    expect(res.body).not.toHaveProperty("passwordHash")
+    expect(res.body).toHaveProperty("email")
+    expect(res.body.email).toBe("testinglowercasing@lowercasing.com")
```
Comment (pbn4) [inline backend/core/test/user/user.e2e-spec.ts:594]: I would be nice to add a login check here, to see if login in with lower cased email works, but that requires you do use admin credentials to confirm a user (set confirmed_at through `PUT /user/:id` endpoint or by setting confirmed_at with UserRepository.

**Comment by YazeedLoonat** [comment]
Hey @pbn4 got this update per your comments, lemme know what you think

**Review by pbn4 — APPROVED** [review]

**Pull request merged**